### PR TITLE
Support StructScan worksheet refs.

### PR DESCRIPTION
**Pull request by lgonzalez-silen**

### CODE REVIEW

**Comment by lgonzalez-silen** [comment]
this deals with simple ref scanning with no cycle checking. will submit another pr for graph checking.

File: marshaling_test.go [hunk 1]
```diff
@@ -266,17 +266,56 @@ func (s *Zuite) TestStructScan_slicesNotSupported() {
 	require.EqualError(s.T(), err, "struct field Texts: cannot StructScan slices (yet)")
 }
 
-type allTypesStruct struct {
-	Ws   *allTypesStruct `ws:"ws"`
-	Num0 int             `ws:"num_0"`
+func (s *Zuite) TestStructScan_refsPtr() {
```
Comment (pascallouisperez) [inline marshaling_test.go]: You'd also need to support cases such as
```
type worksheet self {
  1:me self
  2:my_friends []self
}
```
with graphs of the sort
```
ws.self = ws
ws.my_friends = [friend1, friend2]
friend1.my_friends = [ws, friend2]
friend2.my_friends = [ws, friend1]
```
The pattern followed throughout worksheets is as follows
- https://github.com/helloeave/worksheets/blob/master/marshaling.go#L52
- https://github.com/helloeave/worksheets/blob/master/dbstore.go#L403
- https://github.com/helloeave/worksheets/blob/master/clone.go#L32

Comment (lgonzalez-silen) [reply marshaling_test.go]: Yep, I added a comment above to do cycle checking as a follow up. Happy to add it here if you feel they should go together. Thanks for the references!

**Comment by lgonzalez-silen** [comment]
continued in https://github.com/helloeave/worksheets/pull/115

**Pull request closed without merging**